src/Workspaces/Core/Portable/Shared/Utilities/AsyncBatchingWorkQueue`2.cs

```diff
@@ -161,7 +165,20 @@ void AddItemsToBatch(IEnumerable<TItem> items)
         private ValueTask<TResult> ProcessNextBatchAsync(CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
-            return _processBatchAsync(GetNextBatchAndResetQueue(), _cancellationToken);
+            try
+            {
+                return _processBatchAsync(GetNextBatchAndResetQueue(), _cancellationToken);
+            }
+            catch (Exception ex) when (FatalError.ReportAndPropagateUnlessCanceled(ex, _cancellationToken, ErrorSeverity.Critical))
```

Comment:
Should still avoid breaking the queue here

```suggestion
            catch (Exception ex) when (FatalError.ReportAndCatchUnlessCanceled(ex, _cancellationToken, ErrorSeverity.Critical))
```